File: Source/Front-end/front-end/src/app/model/response/get-luoghi-response.ts
import {Tipo} from "../models/tipo";

export class GetLuoghiResponse {
  id: number;
  nome: string;
  nucleo: string;
  tipo: Tipo;
  piano: number;

  constructor(id: number, nome: string, nucleo: string, tipo: Tipo, piano: number) {
    this.id = id;
    this.nome = nome;
    this.nucleo = nucleo;
    this.tipo = tipo;
    this.piano = piano;

  }
}

File: Source/Back-end/src/main/java/org/elis/manutenzione/dto/response/GetLuoghiResponse.java
package org.elis.manutenzione.dto.response;

import lombok.Data;
import org.elis.manutenzione.model._enum.Tipo;

/**
 * DTO che rappresenta la risposta alla richiesta di ottenere la lista dei luoghi.
 */
@Data
public class GetLuoghiResponse {

    /**
     * L'ID del luogo.
     */
    private long id;

    /**
     * Il nome del luogo.
     */
    private String nome;

    /**
     * Il nucleo a cui appartiene il luogo.
     */
    private String nucleo;

    /**
     * Il tipo di luogo (es. Bagno, Stanza, Luogo Comune).
     */
    private Tipo tipo;

    /**
     * Il piano in cui si trova il luogo.
     */
    private int piano;
}